scripts/data_collection/git_repo.py

```diff
@@ -0,0 +1,46 @@
+import os
+import subprocess
+
+
+class GitRepository:
+    def __init__(self, repository: str, project_dir: str, store_history: bool = False):
+        self.username, self.project_name = repository.split('/')
+        self.project_dir = project_dir
+        self.store_history = store_history
+
+    def clone(self):
+        command = ["git", "clone", f"https://github.com/{self.username}/{self.project_name}.git",
+                   os.path.basename(self.project_dir)]
+        if not self.store_history:
+            command += ["--depth", "1"]
+
+        p = subprocess.Popen(command, cwd=os.path.dirname(self.project_dir))
+        return_code = p.wait()
+        return return_code == 0
+
+    def pull_changes(self) -> bool:
+        """
+        Pull changes from repository with or without history.
+
+        :return: Whether or not repository was updated.
+        """
+        head_before_update = self.get_head_commit()
+        if not self.store_history:
+            p_fetch = subprocess.Popen(["git", "fetch", "--depth", "1"], cwd=self.project_dir, stdout=subprocess.PIPE)
+            p_fetch.wait()
```

Comment:
Maybe should you create a function that receives the necessary args and does these two commands? Currently we can forget about wait and it will produce bugs

Reply:
Also, maybe should you use `subprocess.run`? see [this](https://stackoverflow.com/questions/39187886/what-is-the-difference-between-subprocess-popen-and-subprocess-run#:~:text=run%20executes%20a%20command%20and,receive%20data%20to%20your%20process.) link

In this case, we can create a function like this and don't call `wait`:
```
def run_in_subprocess(command: List[str]) -> str:
    process = subprocess.run(
        command,
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE,
    )

    stdout = process.stdout.decode()
    stderr = process.stderr.decode()

    if stdout:
        logger.debug('%s\'s stdout:\n%s' % (command[0], stdout))
    if stderr:
        logger.debug('%s\'s stderr:\n%s' % (command[0], stderr))

    return stdout
```

Or we can create a class with stdout, stderr and return_code and return it here